HW1/P2/P2b.py

```diff
@@ -1,3 +1,34 @@
 from P2 import *
 
-# Your code here
+# Initialize Spark
+from pyspark import SparkContext
+sc = SparkContext()
+sc.setLogLevel("ERROR")
+
+# Define key parameters
+coordWidth = 2000
+coordHeight = 2000
+noPartitions = 100
+
+# Create RDD with i, j coordinates
+pixCoordIJ = sc.parallelize([(x, y) for x in range(coordWidth) for y in range(coordHeight)])
+
+# Apply the Mandelbrot function & set number of partitions
+pixMandel = pixCoordIJ.map(lambda (i, j): ((i, j), mandelbrot(j/500.0 - 2, i/500.0 - 2))).partitionBy(noPartitions)
```

Comment:
I'm still confused why this helps to balance the workload. I think partitionBy just sorts the RDD by the hash of the keys...you are not guaranteed that this will help balance work. Was the rebalancing simply luck, or am I missing something?